packages/ui/src/views/webhook/webhook-list/repo-webhook-list.tsx

```diff
@@ -0,0 +1,93 @@
+import { Badge, NoData, StackedList } from '@/components'
+
+import { WebhookType } from './types'
+
+export interface PageProps {
+  webhooks?: WebhookType[]
+  LinkComponent: React.ComponentType<{ to: string; children: React.ReactNode }>
+  handleResetFilters?: () => void
+  hasActiveFilters?: boolean
+  query?: string
+  handleResetQuery: () => void
+}
+
+const Title = ({ title, isEnabled }: { title: string; isEnabled: boolean }) => (
+  <div className="inline-flex items-center gap-2.5">
+    {title}
+    <Badge size="sm" disableHover borderRadius="full" theme={isEnabled ? 'success' : 'muted'}>
+      {isEnabled ? 'Enabled' : 'Disabled'}
+    </Badge>
+  </div>
+)
+
+export function RepoWebhookList({
+  webhooks,
+  LinkComponent,
+  handleResetFilters,
+  hasActiveFilters,
+  query,
+  handleResetQuery
+}: PageProps) {
+  return (
+    <>
+      <StackedList.Root>
+        {webhooks && webhooks.length > 0 ? (
```

Comment:
don't use ternary operators for rendering loading/error/no-data states. and don't depend on `length` of data to determine when data is present.
it causes flashing of irrelevant states. instead use short-circuit if conditions, and use the `isLoading`, `isError` states provided by react-query.

Reply:
done